### DISCUSSION TITLE
Further example notebook with useful, standardised plots, groove directory and inprofile query

### DISCUSSION THREAD

**Discussion by sag-kalibreur** [pyroll-project/pyroll-core#208] [Show and tell]
For further inspiration, here is a Jupyter notebook with subdirectories. Groove geometries are imported from the directory grooves if required, so you can build up a kind of database that can be reused. For the inprofile a simple if condition is integrated to flexibly change the shape and size, alternatively these can also be loaded from a subdirectory. With the Marini spread calculation, the spread behaviour can be adapted very well to the real conditions with the coulomb friction coefficient if necessary. The plots show all important process parameters and can be easily reused in other notebooks. The conversion of the SI units (m -> mm, K -> C...) is very simple and is already inserted as a factor in the geometry data or converted in the code of the plots.

[example_ipynb.zip](https://github.com/pyroll-project/pyroll-core/files/15316879/example_ipynb.zip)

**Comment by GRPlan** [comment]
Based on the example provided I have explored some ideas to

-> collect roll data from an Excel workbook (makes use of pandas from_dict plugin)
-> store material data as dictionaries in a separate file and pass them as **kwargs to the in_profile
-> define the in_profile with a wrapper function (should be compatible with interactive input)

all input data are in cell [3]

[ExcelPandas.zip](https://github.com/pyroll-project/pyroll-core/files/15349371/ExcelPandas.zip)

**Reply by sag-kalibreur** [reply]
Great idea, you can enter all the necessary parameters clearly in one cell. The same applies to the excel file with the roll data and the file with the material data. This makes handling easier and more reproducible. In my opinion, this is the right way to go...

**Comment by sag-kalibreur** [comment]
### I have taken up Gerald's additions again and created another sample notebook, this time for a three roll block.

- The rolls/grooves data can be read in from an excel sheet, the material data is in a material file and can also be read in. This is a simple way to use a kind of database

- The parameters for the inprofile and the pass sequence can all be entered clearly in one cell

- The units are converted to the necessary SI units with convert_SI. At the end the pass sequence is generated and calculated

I have tried to design the structure from a user's point of view. Many thanks to @GRPlan for programming the basic code.

As a next step, I would add a query of the inprofile geometry and a query/switch between 2 roll and 3 roll pass sequence. Otherwise a user has to make too many interventions in the code...

**To execute the ipynb, the modules pyroll.from_dict and openpyxl are also required:**

pip install pyroll.from_dict
pip install openpyxl

[sample_excel_matfile_3roll.zip](https://github.com/pyroll-project/pyroll-core/files/15402480/sample_excel_matfile_3roll.zip)

**Comment by sag-kalibreur** [comment]
## Updated sample notebook with 2-roll sequence and 3-roll sequence

### Includes the following main features:

- Loading groove geometries from an excel sheet
- Load material data from a material data file
- Convert SI_units
- Input of required parameters in a single cell
- Mixture of 2-roll passes `and` 3-roll passes 
- The geometry of the Inprofile is calculated manually using a temperature-dependent expansion factor. See [https://github.com/orgs/pyroll-project/discussions/220](url)

The two- and three-roll sequences are combined and calculated at the end.

### Not yet completely solved in this version:

- Transport (stand distance) between the two sequences
- Merging the sequences is not solved nicely

### If not already done additionally install:

`pip install pyroll.from_dict`
`pip install openpyxl`

Try it out.
Post your remarks, comments and possible solutions...

[sample_2roll_3roll_passes_xls_mat.zip](https://github.com/pyroll-project/pyroll-core/files/15424389/sample_2roll_3roll_passes_xls_mat.zip)

**Comment by RollingTechSupport** [comment]
Thank you both @sag-kalibreur  and @GRPlan for your hard work and help to improve the PyRolL - Project. We've used your work and created tests for `pyroll-basic`.

**Reply by sag-kalibreur** [reply]
Very gladly...
